src/main.ts

```diff
@@ -236,11 +230,13 @@ export function activate(context: vscode.ExtensionContext): ReturnType<typeof ge
         }
         if (e && extension.manager.hasTexId(e.document.languageId)) {
             await extension.manager.findRoot()
+            await extension.structureViewer.refreshView()
             extension.linter.lintRootFileIfEnabled()
+        } else if (e && extension.manager.hasBibtexId(e.document.languageId)) {
+            await extension.structureViewer.refreshView()
```

Comment:
We had not better add more lines to `main.ts` if not necessary. How about calling `onDidSaveTextDocument` and `onDidChangeActiveTextEditor` in the constructor of `src/providers/structure.ts` and registering callbacks?

Reply:
What is the general idea behind not adding more lines to `main.ts`? I agree that keeping functions short is a good practice but in this case I am not convinced that spreading subscriptions to `onDidSaveTextDocument` or `onDidChangeActiveTextEditor` in several places will really help.

The `hasBibTeX` part could be moved to a call to `onDidChangeActiveTextEditor` in `src/providers/structure.ts`. In the `hasTexId` part, refreshing the viewer must occur after `findRoot` has completed. It could be moved to `findRoot` but I think that not handling the structure refresh at the same place for LaTeX and BibTeX files will make maintenance harder.

Reply:
If we put everything into `main.ts`, we must always consider and remember all the things when we edit `main.ts`. Even if we can easily remember the reason of the conditions now, we would not be able to do a few weeks later.

Reply:
> In the hasTexId part, refreshing the viewer must occur after findRoot has completed.

I have missed that. It is not easy to see the dependency. At least, you should write it in a comment.

Reply:
Apart from the above discussion, the change of the active text editor does not necessarily mean the change of the rootfile. Should we really always call `structureViewer.refreshView()`?

Reply:
This is indeed the whole point of the PR. `structureViewer.refreshView` calls `SectionNodeProvider.build(force=false)`

https://github.com/James-Yu/LaTeX-Workshop/blob/2fe4d967aa7f07e58d6d1f5cee06f187f14eb754/src/providers/structure.ts#L57-L60

The structure is really rebuilt only when the `rootFile` changes

https://github.com/James-Yu/LaTeX-Workshop/blob/2fe4d967aa7f07e58d6d1f5cee06f187f14eb754/src/components/manager.ts#L377

Reply:
I must say it is not easy to see the difference between`update` and `refreshView` from the names. The should have much more different names.

Reply:
Any suggestions?  What about `updateData` and `updateView`?

Reply:
They should not have similar names. They should not have the same word. How about `constructTreeStructure` and `refreshView`